### ISSUE TITLE
[Feature] Move all portable data directories into a single sub directory

### ISSUE THREAD

**Issue by mgziminsky** [WerWolv/ImHex#659]
### What feature would you like to see?

In portable mode, it would be better to have a single `userData` sub directory that contains all of the other data directories. If this is acceptable, I can get a dev environment set up when i get a chance and submit a PR if necessary.

### How will this feature be useful to you and others?

Having all the data directories in the root application folder makes it difficult to distinguish between user data and base application data. It also makes it more difficult to create backups and to update, since it isn't obvious what came as part of the base install vs what was added by the user.

Having a single user sub directory makes managing a portable installation much more simple and clean. This has the following primary benefits:
- Backups can be done on a single directory. So any future additions/removals will automatically included.
- Updating only needs to preserver a single directory. Copy the one directory to a new install and be done. No need to decipher what came pre-installed and what was user added.

### Request Type

- [X] I can provide a PoC for this feature or am willing to work on it myself and submit a PR

**Comment by WerWolv** [comment]
Hey
There's actually not really a difference between user data and base application data. ImHex just ships with the latest versions of patterns, libraries and so on but they are meant to be extended or updated with the content store.
I can move all the folders from next to the exe into a subfolder but it won't change much in my opinion

**Comment by mgziminsky** [comment]
It doesn't change much in terms of actual application usability, but makes a big difference organizationally. Your comment regarding extending what's shipped is the main factor. If I create my own patterns, or download any from a 3p, it is preferable to have them stored separately from the defaults. This is especially true if I decided to modify one of the defaults for some reason, since it wouldn't get overwritten on upgrade if it was stored separately.

I'm not asking to "move" all the existing default files, they should stay where they are, just that there be an additional lookup path in a single sub folder, same as it works in non portable mode that uses a system config path, but with a local path instead.


Thanks for considering and for your work. I used ImHex recently during a project of mine to help RE a file format, and overall it worked great. There are definitely some little kinks and things that need ironed out, but I can see you are actively working on it. Your last batch of commits actually just fixed a little usability issue I was considering making a PR for (selection size in hex). So thanks again for your work.
